runtime/stdlib/test.go

```diff
@@ -349,33 +263,186 @@ func init() {
 		),
 	)
 
-	// Test.readFile()
-	testContractType.Members.Set(
-		testReadFileFunctionName,
-		sema.NewUnmeteredPublicFunctionMember(
-			testContractType,
-			testReadFileFunctionName,
-			testReadFileFunctionType,
-			testReadFileFunctionDocString,
-		),
-	)
+	blockchainType, ok := testContractType.NestedTypes.Get(blockchainTypeName)
+	if !ok {
+		panic(typeNotFoundError(testContractTypeName, blockchainTypeName))
+	}
+
+	testNewEmulatorBlockchainFunctionType = &sema.FunctionType{
+		ReturnTypeAnnotation: sema.NewTypeAnnotation(
+			blockchainType,
+		),
+	}
+
+	initTestContractTypeFunctions()
+
+	// Enrich 'Test' contract elaboration with natively implemented composite types.
+	// e.g: 'EmulatorBackend' type.
+	testContractChecker.Elaboration.SetCompositeType(
+		testEmulatorBackend.compositeType.ID(),
+		testEmulatorBackend.compositeType,
+	)
+}
+
+func initBlockchainBackendInterfaceType() *sema.InterfaceType {
+	typ, ok := testContractType.NestedTypes.Get(blockchainBackendTypeName)
+	if !ok {
+		panic(typeNotFoundError(testContractTypeName, blockchainBackendTypeName))
+	}
+
+	blockchainBackendInterfaceType, ok := typ.(*sema.InterfaceType)
+	if !ok {
+		panic(errors.NewUnexpectedError(
+			"invalid type for '%s'. expected interface",
+			blockchainBackendTypeName,
+		))
+	}
+	return blockchainBackendInterfaceType
+}
+
+func initMatcherType() *sema.CompositeType {
+	typ, ok := testContractType.NestedTypes.Get(matcherTypeName)
```

Comment:
Here and elsewhere: Instead of accessing the global here (in this case, `testContractType`), pass it as a parameter